docs/architecture.md

```diff
@@ -1,188 +1,260 @@
-dictates that all the Webhook endpoints must be secured with TLS.
-`kubewarden-controller` takes care of setting up this secure communication
-by doing these steps:
+Kubewarden works as a Kubernetes Admission Webhook.
+Kubernetes specifies using TLS to secure all Webhook endpoints.
+The `kubewarden-controller` sets up this secure communication
+by:
+
+1. Generating a self-signed Certificate Authority
+1. Use this CA to generate a TLS certificate key for the `policy-server` Service.
+
+<!--The capitalization on Secret is ok? Maybe in ``?-->
 
-1. Generate a self-signed Certificate Authority
-1. Use this CA to generate a TLS certificate and a TLS key for the
-   `policy-server` Service.
+These objects are all stored as Secret resources in Kubernetes.
 
-All these objects are stored into Kubernetes as Secret resources.
+<!--TODO:Next paragraph. `policy-server` is in the aquamarine box, right? I
+don't think the K8s ClusterIP Service is shown in the diagram anywhere right?
+Should it be?-->
 
-Finally, `kubewarden-controller` will create the `policy-server`
-Deployment and a Kubernetes ClusterIP Service to expose it inside of
-the cluster network.
+Finally, `kubewarden-controller` creates the `policy-server` Deployment
+and a Kubernetes ClusterIP Service
+to expose it inside the cluster network.
 
 ### Defining the first policy
 
-This chart shows what happens when the first policy bounded to the default `policy-server` is defined inside of the
-cluster:
+<!--TODO:First mention of 'bound'. I'd like to define what binding or to
+be bound is, how it happens, where, etc?-->
+
+This diagram shows what happens when defining the first policy
+bound to the default `policy-server` in the cluster:
 
 ![Defining the first ClusterAdmissionPolicy resource](/img/architecture_sequence_02.png)
 
-`kubewarden-controller` notices the new `ClusterAdmissionPolicy` resource and,
-as a result of that, it finds the bounded `PolicyServer` and reconciles it.
+<!--TODO:There is ClusterAdmissionPolicy inside and outside the API
+Server box.  What is this signifying? And for PolicyServer? Definition
+of reconcile? Just to find the PolicyServer that matches the
+ClusterAdmissionPolicy?-->
+
+The `kubewarden-controller` notices the new `ClusterAdmissionPolicy` resource and
+so finds the bound `PolicyServer` and reconciles it.
 
 ### Reconciliation of `policy-server`
 
-When a `ClusterAdmissionPolicy` is created, modified or deleted a reconciliation loop for the `PolicyServer`
-that owns the policy is triggered inside the `kubewarden-controller`.
-In this reconciliation loop, a ConfigMap with all the polices bounded to 
-the `PolicyServer` is created. Then the a Deployment rollout of the
-interested `policy-server` is started. As a result of that, the new `policy-server`
-instance will be started with the updated configuration.
+<!--TODO:'reconciliation loop', can we show this loop in the diagram
+somehow? ConfigMap, should be in the diagram? In the OCI registry?-->
```

Comment:
I think is doable. But I guess we need to change the diagram, and maybe the kind of it, to show this logic properly.

Reply:
it's doable, but it's a bit of work tbh. It means making diagrams for the code in kubewarden-controller, which are more specific and get out of sync. I'm ok just saying "reconciling" here.

Reply:
ok, thanks.